src/legacy/core_plugins/kibana/public/dashboard/dashboard_app.js

```diff
@@ -59,6 +58,8 @@ import { getUnhashableStatesProvider } from 'ui/state_management/state_hashing';
 
 import { DashboardViewportProvider } from './viewport/dashboard_viewport_provider';
 
+import 'plugins/data';
```

Comment:
I was also bothered by the implicit import here.
Love your suggestion!

Reply:
Since there are going to be multiple loadLegacyDirectives in many files (per plugin), I'm also going to add a convenience method on the service itself.

```

import { data } from 'plugins/data';
data.loadLegacyDirectives();

```

Reply:
Yeah that would work too - the only downside I can think of is that by making the services implicit, we'll just have to reverse-engineer all references to `data.loadLegacyDirectives()` later to figure out exactly what services' directives a particular module is relying on.

If we are already doing the work today of pulling directives into specific services and updating downstream imports, keeping everything explicit might save us from more work later (avoiding an "autoload" type situation where you don't know what a module's true dependencies are).

But I'd defer to you on this -- you've spent much more time dealing with various legacy angular directives at this point 😃

Reply:
I found a nice (and necessary) middle ground, which you can see in the filter bar PR
https://github.com/elastic/kibana/pull/35460/files#diff-23034bf5cdc67a7321dbfb230edacd84